source/fem-dem/vans_assemblers.cc

```diff
@@ -422,10 +422,16 @@ GLSVansAssemblerBDF<dim>::assemble_rhs(
                ++p)
             {
               local_rhs_i -=
-                (bdf_coefs[p] * void_fraction[0] * (velocity[p] * phi_u_i) +
-                 bdf_coefs[p] * (void_fraction[p] * scratch_data.phi_p[q][i]));
+                (bdf_coefs[p] * void_fraction[0] * (velocity[p] * phi_u_i));
 
-              if (cfd_dem.grad_div == true)
+              if (cfd_dem.void_fraction_time_derivative == true)
+                {
+                  local_rhs_i -= bdf_coefs[p] *
+                                 (void_fraction[p] * scratch_data.phi_p[q][i]);
+                }
+
+              if (cfd_dem.grad_div == true &&
+                  cfd_dem.void_fraction_time_derivative == true)
```

Comment:
You can put this if in the if above no? this would make things easier to read I think.

Reply:
done